src/node_url_pattern.cc

```diff
@@ -0,0 +1,571 @@
+#include "node_url_pattern.h"
+#include "base_object-inl.h"
+#include "debug_utils-inl.h"
+#include "env-inl.h"
+#include "memory_tracker-inl.h"
+#include "node.h"
+#include "node_errors.h"
+#include "node_mem-inl.h"
+#include "path.h"
+#include "util-inl.h"
+
+namespace node::url_pattern {
+
+using v8::Array;
+using v8::Boolean;
+using v8::Context;
+using v8::DontDelete;
+using v8::Function;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::IndexedPropertyHandlerConfiguration;
+using v8::Integer;
+using v8::Intercepted;
+using v8::Isolate;
+using v8::JustVoid;
+using v8::Local;
+using v8::Map;
+using v8::Maybe;
+using v8::MaybeLocal;
+using v8::Name;
+using v8::NamedPropertyHandlerConfiguration;
+using v8::NewStringType;
+using v8::Nothing;
+using v8::Null;
+using v8::Object;
+using v8::PropertyAttribute;
+using v8::PropertyCallbackInfo;
+using v8::PropertyDescriptor;
+using v8::PropertyHandlerFlags;
+using v8::ReadOnly;
+using v8::String;
+using v8::Uint32;
+using v8::Value;
+
+URLPattern::URLPattern(Environment* env,
+                       Local<Object> object,
+                       ada::url_pattern&& url_pattern)
+    : BaseObject(env, object) {
+  MakeWeak();
+  url_pattern_ = std::move(url_pattern);
+}
+
+void URLPattern::MemoryInfo(MemoryTracker* tracker) const {
+  // TODO: Implement this.
+}
+
+void URLPattern::New(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args.IsConstructCall());
```

Comment:
since this constructor is exposed directly to users, this should throw an exception rather than abort

Reply:
```suggestion
  if (!args.IsConstructCall()) {
    return THROW_........
  }
```

Replace the CHECK here with an appropriate error.